### PR TITLE
Attempt to purge all vaults, managed HSMs

### PR DESCRIPTION
**Pull request by heaths**
Reverts #1910. Vaults and managed HSMs are automatically purged on their purge date. The point was to purge them daily to preserve capacity. The default purge date is +90 days.

### CODE REVIEW

File: eng/scripts/live-test-resource-cleanup.ps1
Comment (weshaggard) [inline eng/scripts/live-test-resource-cleanup.ps1:70]:
```suggestion
# Get purgeable resources already in a deleted state.
```
Any reason to do this before cleaning up the resource groups? I moved it down because I thought it would allow the resource group clean-up to finish before it even tries to retrieve these.

File: eng/common/scripts/Helpers/Resource-Helpers.ps1
Comment (weshaggard) [inline eng/common/scripts/Helpers/Resource-Helpers.ps1:10]: It might not be worth pushing this change to all the other repos but if you want to do it locally until we get through the list that is fine but I leave it up to you.

File: eng/scripts/live-test-resource-cleanup.ps1
Comment (heaths) [reply eng/scripts/live-test-resource-cleanup.ps1:70]: It'd be redundant. `Get-PurgeableGroupResources` will get them, so they'd be double-counted. `Get-PurgeableResources` only gets *deleted* resources. The expectation is that, normally, KVs and MHSMs in a group that isn't deleted also aren't yet deleted.

File: eng/common/scripts/Helpers/Resource-Helpers.ps1
Comment (heaths) [inline eng/common/scripts/Helpers/Resource-Helpers.ps1:178]: I have a different changing using `$using:` here but don't want to push it yet because it will cancel the 6+ hour job deleting the remaining KVs from the Test Resources sub. Still more subs to go, but may cancel it after that to test those changes. I ran them locally and it seems to work fine.

File: eng/common/scripts/Helpers/Resource-Helpers.ps1
Comment (weshaggard) [inline eng/common/scripts/Helpers/Resource-Helpers.ps1:119]: nit: just for readability it might be worth splitting the script block onto a newline like you did with the other case.

File: eng/common/scripts/Helpers/Resource-Helpers.ps1
Comment (weshaggard) [inline eng/common/scripts/Helpers/Resource-Helpers.ps1:160]: Do you still need to pass thru the resource?

Comment (weshaggard) [reply eng/common/scripts/Helpers/Resource-Helpers.ps1:160]: Nevermind I see what you used that for now.

**Review by weshaggard — APPROVED** [review]
The changes looks good. Lets wait until we get through the backlog before merging and we undo the pipeline timeout change.

File: eng/common/scripts/Helpers/Resource-Helpers.ps1
Comment (heaths) [reply eng/common/scripts/Helpers/Resource-Helpers.ps1:160]: It's used for printing a warning that the operation timed out, as well as optionally outputting all timed-out resources at the end so we can investigate if necessary.

File: eng/common/scripts/Helpers/Resource-Helpers.ps1
Comment (heaths) [reply eng/common/scripts/Helpers/Resource-Helpers.ps1:119]: I could. Originally it was because I was also logging (as warning) the failed script and this one-liner looked nicer, but I'm no longer doing that. That said, I may only do it if there's any further changes and...I really hope not. This has taken much more time than I anticipated, and not even just the execution of it. The closure issue was a real head-scratcher.

Comment (heaths) [reply eng/common/scripts/Helpers/Resource-Helpers.ps1:119]: Done

**Comment by heaths** [comment]
I'm proceeding with the eng/common merge.

**Pull request merged**
